### DISCUSSION TITLE
Best Practice get Tenant in Queue Jobs

### DISCUSSION THREAD

**Discussion by jappi00** [filamentphp/filament#9085] [Help]
Hello together,

In the past I have wrote my whole app to work in sync. Now I want to give the user the option to run some task asynchronous in a queue. Is there a way i can set the return value of Filament::getTenant()? In the far feature I want to trigger some tasks from an Cron aswell. What is here the best practice to set the tenant without an user?

Maybe somebody has an good idea? thanks for every input!

**Comment by jappi00** [comment]
Maybe there is a way i didn't see similar to this:
https://tenancyforlaravel.com/docs/v3/queues

**Reply by monzer15** [reply]
Hello, i have the same issue Filament::getTenant() returns null in any queued job, did u find a fix for this?

**Comment by juliangums** [comment]
@jappi00 @monzer15 sounds like basic job dispatching (https://laravel.com/docs/11.x/queues#dispatching-jobs). 
You just add a parameter for it in your job class and then pass it where you dispatch it:
```php
$tenant = Filament::getTenant();
YourJob::dispatch($someOtherData, $tenant);
```

Your job class should look something like this
```php
class MyJob implements ShouldQueue
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;

    protected $someOtherData;
    
    protected $tenant;

    public function __construct($someOtherData, $tenant)
    {
        $this->someOtherData = $someOtherData;
        $this->tenant = $tenant;
    }

    public function handle()
    {
        // do something with the tenant here
        $tenant = $this->tenant;
    }
}
```

**Reply by monzer15** [reply]
ok thanks!
